### DISCUSSION TITLE
[Request] Modify Hit Delay

### DISCUSSION THREAD

**Discussion by TheWranger** [Zrips/CMI#1332] [Feature Requests]
**Description of issue or feature request:**
**This isn't regarding the 1.9+ hit cooldowns**
Minecraft has a default hit delay of 20 ticks but it can lead to smoother combat to lower this by a few ticks (this is what servers with "Good Hit Detection" do) - it would be nice if we had some way to modify this delay within CMI without having to use other plugins. Many of the other plugins that do this are either bloated with other features or are no longer being updated.

**Comment by Drc-DEV** [comment]
It would be cool to have an overall solution for changing pvp behaviour. Somewhat similar to https://www.spigotmc.org/resources/knockbackmaster-⭐-professional-custom-knockback-and-hits-for-pvp.42721/

Hit Delay
Hit Cooldown
Knockback

**Comment by TheWranger** [comment]
What do you mean by Hit Cooldown? OldCombatMechanics does fine at removing hit cooldowns.

**Comment by Drc-DEV** [comment]
I mean, having an All-In-One hit configuration system would be cool. So that plugins like OldCombatMechanics, FastPot or NoHitDelay are not needed.

```
FastPot: 1.2 (Potions are thrown with more velocity)
AttackSpeed: 24 (The attackspeed attribute) (p.getAttribute(Attribute.GENERIC_ATTACK_SPEED).setBaseValue(24))
HitDelay: 8 (delay between two hits) (p.setMaximumNoDamageTicks(8))
Knockback: (Custom knockback management)
    Horizontal: 0.31
    Vertical: 0.21
    SprintH: 1.3
    SprintY: 0.8
PlayerCollisions: false (Toggling player collisions)
```

Something like this.

**Comment by TheWranger** [comment]
I still really don't see the need to have a replacement for OldCombatMechanics built-in because of how well it does its job and how it isn't a small plugin like FastPot or NoHitDelay are.

**Comment by Drc-DEV** [comment]
Then you should ask the developer of OldCombatMechanics to include HitDelay, as that would be more appropriate.
https://github.com/gvlfm78/BukkitOldCombatMechanics/issues/19

**Comment by Drc-DEV** [comment]
Anyways OldCombatMechanics has a lot of bugs for me 1.13.2, so having a stable and well-done all in one pvp management system inside of CMI would be a better solution for me at least.

**Comment by kernitus** [comment]
@Drc-DEV If there are bugs you should be reporting them, cause I can't fix them otherwise...
